pkg/client/unversioned/helper.go

```diff
@@ -290,7 +291,8 @@ func SetKubernetesDefaults(config *Config) error {
 	if len(config.UserAgent) == 0 {
 		config.UserAgent = DefaultKubernetesUserAgent()
 	}
-	if len(config.Version) == 0 {
+	if len(config.Version) == 0 || apiutil.GetGroup(config.Version) != "" {
```

Comment:
Is this modification required in this PR? If not, I can fix it in my PR which deals with clients and their configs.